omod/src/main/webapp/resources/htmlforms/072-IntegratedMaternityRegister.xml

```diff
@@ -11,203 +11,352 @@
-                    $hivInformationSection.find('input').removeAttr('disabled');
-                    $arvsAdministeredToBabyField.removeAttr('disabled');
-					$hivInformationSection.find('input,select').removeAttr('disabled');
-                } else {
-					$hivInformationSection.find('input,select').each(function() {										
-						if (jq(this).attr('id') != $hivStatusField.attr('id')) {
-							jq(this).attr('disabled', true);
-						} else {
-							jq(this).removeAttr('disabled');
-						}
-					});
-                    $arvsAdministeredToBabyField.attr('disabled', true);
-                }
-				viralLoadResultsChange();
-            };
-
-            var iycfChange = function() {
-                var $iycf =  getValue('iycf.value');
-                var $infantFeedingOptionField = getField('infant-feeding-option.value');
-                if ($iycf == YES_CONCEPT_ID) {
-                    $infantFeedingOptionField.removeAttr('disabled');
-                } else {
-                    $infantFeedingOptionField.attr('disabled', true);
-                }
-            };
-
-            var arvsToMotherChange =  function() {
-                var $arvsToMother =  getValue('arvs-to-mother.value');
-                var $preArtNumberField = jq(PRE_ART_NUMBER_FIELD).find('input:text');
-                if ($arvsToMother != '') {
-                    $preArtNumberField.removeAttr('disabled');
-                } else {
-                    $preArtNumberField.attr('disabled', true);
-                }
-            };
-
-            $viralLoadResultsField.change(viralLoadResultsChange);
-            $hivStatusField.change(hivStatusChange);
-            $iycfField.change(iycfChange);
-            $arvsToMotherField.change(arvsToMotherChange);
-			
-			viralLoadResultsChange();
-			hivStatusChange();
-			iycfChange();
-			arvsToMotherChange();
-
-        });
-
-        if (jQuery) {
-            jQuery(document).ready(function (jq) {
+	    jQuery(document).ready(function(jq){
+	        const VIRAL_LOAD_RESULTS_CONCEPT_ID = 1305;
+	        const VIRAL_LOAD_DETECTED_CONCEPT_ID = 1301;
+	        const VIRAL_LOAD_NOT_DETECTED_CONCEPT_ID = 1306;
+	        const VIRAL_LOAD_SAMPLE_REJECTED_CONCEPT_ID = 1304;
+
+	        const HIV_POSITIVE_CONCEPT_ID = 90166;
+	        const HIV_NEGATIVE_CONCEPT_ID = 90167;
+
+	        const YES_CONCEPT_ID = 1065;
+	        const NO_CONCEPT_ID = 1066;
+
+	        const HIV_INFORMATION_SECTION = '#hiv-information-section';
+	        const PRE_ART_NUMBER_FIELD = '#pre-art-number';
+	        const ARVS_ADMINISTERED_TO_BABY_FIELD = '#arvs-administered-to-baby';
+	        const IYCF_FIELD = '#iycf';
+
+	        var $hivStatusField = getField('hiv-status.value');
+	        var $viralLoadResultsField = getField('viral-load-results.value');
+	        var $iycfField = jq(IYCF_FIELD).find('input');
+	        var $arvsToMotherField = getField('arvs-to-mother.value');
+
+	        var common = {

```

Comment:
Why introduce this function over:

1. OpenMRS - https://github.com/openmrs/openmrs-core/blob/1.11.x/webapp/src/main/webapp/openmrs.js

2. UgandaEMR - https://github.com/METS-Programme/openmrs-module-aijar/blob/master/omod/src/main/webapp/resources/js/aijar.js 

If we need helper fields let us build on exisiting ones and also build functions that are reusable across forms